sensors/listener.go

```diff
@@ -331,7 +331,11 @@ func (sensorCtx *SensorContext) triggerActions(ctx context.Context, sensor *v1al
 		depNames = append(depNames, k)
 		eventIDs = append(eventIDs, v.ID())
 	}
-	go sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
+	if trigger.AtLeastOnce {
+		sensorCtx.triggerWithRateLimit(ctx, sensor, trigger, eventsMapping, depNames, eventIDs)
```

Comment:
My understanding this is block processing, how is it related to `atLeastOnce`?

Reply:
A function higher up in the call chain acks the message in a defer statement. If `sensorCtx.triggerWithRateLimit` is called in go routine, then we ack the message without waiting to see if the trigger invocation was successful or unsuccessful, this is equivalent of at most once. If `sensorCtx.triggerWithRateLimit` is called synchronously (aka not in a go routine) then we wait to ack the message until know the trigger has been invoked (or we have exhausted the retry loop).

Reply:
@bilalba maybe we should add an explanatory comment here?